distributed/scheduler.py

```diff
@@ -1050,19 +1032,18 @@ def add_exec_time(self, duration: float) -> None:
             self.duration_average = -1
 
     def add_duration(self, action: str, start: float, stop: float) -> None:
-        super().add_duration(action, start, stop)
-        duration_s = self._calculate_duration_us(start, stop) / 1e6
+        duration_us = max(round((stop - start) * 1e6), 0)
+        self._duration_us += duration_us
+        self._all_durations_us[action] += duration_us
```

Comment:
this contradicts DRY but I think it's worth it considering that it's just two places and the complexity is low